internal/rbd/rbd_util.go

```diff
@@ -135,8 +136,27 @@ type rbdSnapshot struct {
 	RequestName    string
 }
 
+// imageFeature represents required image features and value.
+type imageFeature struct {
+	// needRbdNbd indicates whether this image feature requires an rbd-nbd mounter
+	needRbdNbd bool
```

Comment:
Are these features really only available when using rbd-nbd? Or does it depend on the kernel version? @dillaman

Reply:
Journaling is never supported with the kernel. You would have to go pretty far back for a kernel that doesn't support layering / exclusive-lock. The object-map/fast-diff/deep-flatten features are another story. It's all documented here [1] but your mileage may vary when dealing w/ RHEL/CentOS due to backporting.

[1] https://docs.ceph.com/en/latest/rbd/rbd-config-ref/#image-features